show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,259 @@
+
+    // Create a graphic near the Empire State Building to be the observer
+    private val observerGraphic = Graphic(
+        geometry = Point(
+            x = -73.9853,
+            y = 40.7484,
+            z = 50.0,
+            spatialReference = SpatialReference.wgs84()
+        ),
+        symbol = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Circle,
+            color = Color.red,
+            size = 5f
+        )
+    )
+
+    // Create a ViewModel to handle dialog interactions
+    private val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    init {
+
+        // Zoom to show the observer
+        val camera = Camera(
+            observerGraphic.geometry as Point,
+            distance = 700.0,
+            roll = -30.0,
+            pitch = 45.0,
+            heading = 0.0,
+        )
+
+        // Define base surface for elevation data
+        val surface = Surface().apply {
+            elevationSources.add(
+                ArcGISTiledElevationSource(
+                    uri = getString(
+                        application,
+                        R.string.elevation_service_url
+                    )
+                )
+            )
+        }
+
+        // Define a scene layer for the New York buildings
+        val buildings =
+            ArcGISSceneLayer(uri = application.getString(R.string.new_york_buildings_service_url))
+
+        // Add the surface and buildings to the scene, and define the viewpoint on launch
+        scene.apply {
+            baseSurface = surface
+            operationalLayers.add(buildings)
+            initialViewpoint = Viewpoint(
+                boundingGeometry = observerGraphic.geometry as Point,
+                camera = camera
+            )
+        }
+
+        // Set up a heading expression to handle graphic rotation
+        val renderer3D = SimpleRenderer().apply {
+            sceneProperties.headingExpression = ("[HEADING]")
+        }
+
+        // Add the graphics to the graphic overlay
+        graphicsOverlay.apply {
+            sceneProperties.surfacePlacement = SurfacePlacement.RelativeToScene
+            graphicsOverlay.graphics.add(observerGraphic)
+            graphicsOverlay.graphics.add(taxiGraphic)
+            graphicsOverlay.renderer = renderer3D
+        }
+
+        viewModelScope.launch {
+
+            scene.load().onSuccess {
+
+                // Create a line of sight between the two graphics and add it to the analysis overlay
+                val lineOfSight = GeoElementLineOfSight(
+                    observerGeoElement = observerGraphic,
+                    targetGeoElement = taxiGraphic
+                )
+                analysisOverlay.analyses.add(lineOfSight)
+
+                // Select (highlight) the taxi when the line of sight target visibility changes to visible
+                if (lineOfSight.isVisible) {
+                    taxiGraphic.isSelected = lineOfSight.isVisible
+                }
```

Comment:
None of these are really suspend calls, and I'm wondering if this is needed to be in a coroutine scope, or even on scene loaded.